index.js

```diff
@@ -339,3 +339,65 @@ module.exports.random = {
     return s * Math.pow(2, (-64 - e))
   }
 }
+
+/**
+ * Utils for simple signing and verification of HTTPS requests
+ **/
+module.exports.https = {
+  /**
+   * Sign the `message` with the given `secretKey`.
+   *
+   * @method
+   * @param {any} headers - headers containing the properties to sign
+   * @param {string} keyId - an opaque string that the server can use to look up the component they need to validate the signature
+   * @param {string} secretKey - secret key to sign the message
+   * @returns {string}
+   */
+  ed25519Sign: function (headers /* any */, keyId /* string */, secretKey /* string */) {
+    if (!secretKey) throw new Error('secret key is required')
+    if (!keyId) throw new Error('key id is required')
+    if (JSON.stringify(headers) === '{}') throw new Error('headers are required')
+
+    const headerKeys = []
+    const message = Object.entries(headers)
+      .map(([key, value]) => {
+        headerKeys.push(key)
+        return `${key}: ${value}`
+      }).join('\n')
+
+    // Generate signature using the ed25519 algorithm.
+    const sign = nacl.sign.detached(
+      Uint8Array.from(Buffer.from(message)),
+      Uint8Array.from(Buffer.from(secretKey, 'hex'))
+    )
+
+    const signature = Buffer.from(sign).toString('base64')
+    return `keyId="${keyId}",algorithm="ed25519",headers="${headerKeys.join(' ')}",signature="${signature}"`
+  },
+
+  /**
+   * Verify the signed `message` with the given `publicKey`.
+   *
+   * @method
+   * @param {any} headers - headers containing the signature for verification
```

Comment:
this should be annotated as Object and headers.signature should be annotated as well. see "parameters with properties" example in https://jsdoc.app/tags-param